packages/@okta/vuepress-site/code/react/okta_react/index.md

```diff
@@ -39,31 +39,89 @@ A simple way to add authentication to a React app is using the [Okta Auth JS](/c
 npm install @okta/okta-auth-js --save
 ```
 
-We'll also need `@okta/okta-react` and `react-router-dom` to manage our routes:
+We'll also need `@okta/okta-react` and `react-router-dom` to manage our routes (`@okta/okta-react` can be used to support other router libraries, but `react-router-dom` has pre-existing support).
 ```bash
 npm install @okta/okta-react react-router-dom --save
 ```
 
 ## Create a Custom Login Form
-If the [Okta Sign-In Widget](/code/javascript/okta_sign-in_widget/) does not fit your needs, [Auth-JS](/code/javascript/okta_auth_sdk/) provides lower-level access to User Lifecycle operations, MFA, and more. For this example, we'll create a simple username and password form without MFA.
+If the [Okta Sign-In Widget](/code/javascript/okta_sign-in_widget/) does not fit your needs, [AuthJS](/code/javascript/okta_auth_sdk/) provides lower-level access to User Lifecycle operations, MFA, and more. For this example, we'll create a simple username and password form without MFA.
 
-Create a `src/LoginForm.js` file:
+Create a `src/LoginForm.jsx` file:
 
+`src/LoginForm.jsx` using a function-based component:
+
+```jsx
+// src/LoginForm.jsx
+
+import React, { useState } from 'react';
+import OktaAuth from '@okta/okta-auth-js';
+import { useOktaAuth } from '@okta/okta-react';
+
+const LoginForm = ({ baseUrl }) => { 
+  const { authService } = useOktaAuth();
+  const [sessionToken, setSessionToken] = useState();
+  const [username, setUsername] = useState();
+  const [password, setPassword] = useState();
+  
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    const oktaAuth = new OktaAuth({ url: baseUrl });
+    oktaAuth.signIn({ username, password })
+      .then(res => setSessionToken(res.sessionToken))
+      .catch(err => console.log('Found an error', err));
+  };
+
+  const handleUsernameChange = (e) => {
+    setUsername(e.target.value);
+  };
+
+  const handlePasswordChange = (e) => {
+    setPassword(e.target.value);
+  };
+
+  if (sessionToken) {
+    authService.redirect({ sessionToken });
+    return null;
+  }
+
+  return (
+    <form onSubmit={handleSubmit}>
+      <label>
```

Comment:
Two <label> tags instead of one?

Reply:
Ah good catch - this was in the original, fixed now.